### PR TITLE
Move readiness probe script to configmap

### PR DESCRIPTION
**Pull request by david-kow**
Moving readiness probe script from an inline script to config map. Part of the work for #932.
1. configmap gets created. This happens in Reconcile function, as configmap is shared between pods.
1. volume with scripts is added to pod spec.

configmap/configmap.go, configmap_control.go were revived since they are helpful here (@nkvoll you removed them previously, so let me know if I should move that code somewhere else).

### CODE REVIEW

File: operators/pkg/controller/elasticsearch/volume/volume.go [hunk 1]
```diff
@@ -36,6 +36,9 @@ const (
 
 	ElasticsearchDataVolumeName = "elasticsearch-data"
 	ElasticsearchLogsVolumeName = "elasticsearch-logs"
+
+	ScriptsVolumeName      = "elastic-internal-scripts"
+	ScriptsVolumeMountPath = "/mnt/elastic/scripts"
```
Comment (thbkrkr) [inline operators/pkg/controller/elasticsearch/volume/volume.go]: With the new naming this should be `/mnt/elastic-internal/scripts`.

Comment (david-kow) [reply operators/pkg/controller/elasticsearch/volume/volume.go]: Thanks, fixed.

File: operators/pkg/controller/elasticsearch/volume/configmap.go [hunk 2]
```diff
@@ -10,10 +10,16 @@ import (
 
 // NewConfigMapVolume creates a new ConfigMapVolume struct
 func NewConfigMapVolume(configMapName, name, mountPath string) ConfigMapVolume {
+	return NewConfigMapVolumeWithMode(configMapName, name, mountPath, corev1.ConfigMapVolumeSourceDefaultMode)
+}
+
+// NewConfigMapVolume creates a new ConfigMapVolume struct with default mode
+func NewConfigMapVolumeWithMode(configMapName, name, mountPath string, defaultMode int32) ConfigMapVolume {
```
Comment (barkbay) [inline operators/pkg/controller/elasticsearch/volume/configmap.go]: godoc not in sync

File: operators/pkg/controller/elasticsearch/configmap/configmap_control.go [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+package configmap
+
+import (
+	"reflect"
+
+	"github.com/elastic/cloud-on-k8s/operators/pkg/apis/elasticsearch/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/controller/common/reconciler"
+	"github.com/elastic/cloud-on-k8s/operators/pkg/utils/k8s"
+
```
Comment (barkbay) [inline operators/pkg/controller/elasticsearch/configmap/configmap_control.go]: We try to keep only two groups in imports: std libs and third parties. But it is not easy if you are using IntelliJ ;)

File: operators/pkg/controller/elasticsearch/volume/configmap.go (hunk 2 above)
Comment (david-kow) [reply operators/pkg/controller/elasticsearch/volume/configmap.go]: Ah, nice catch, fixed.

File: operators/pkg/controller/elasticsearch/configmap/configmap_control.go (hunk 3 above)
Comment (david-kow) [reply operators/pkg/controller/elasticsearch/configmap/configmap_control.go]: Fixed.

**Pull request merged**